src/main/scala/is/hail/linalg/BlockMatrix.scala

```diff
@@ -1083,6 +1082,27 @@ class BlockMatrix(val blocks: RDD[((Int, Int), BDM[Double])],
       }, name, reqDense = reqDense)
   }
 
+  def reduce(blockOp: BDM[Double] => Double, scalarOp: (Double, Double) => Double): Double =
```

Comment:
seems odd from an interface perspective to not expose the zero value as a parameter.

Reply:
0.0 is also the right zero for extensions like min, max, and prod since all blocks missing means all entries zero (with at least one entry), so i'm going to leave it for now